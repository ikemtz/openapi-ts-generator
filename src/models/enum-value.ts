export interface IEnumValue {
  key?: number;
  name: string;
  titleName: string;
  snakeCaseName: string;
}
