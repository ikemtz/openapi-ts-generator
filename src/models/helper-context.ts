import { IEntity } from './template-data';

export interface IHelperContext {
  data: {
    root: IEntity;
    key: number;
    index: number;
    first: boolean;
  };
  name: string;
}

export type PropertyType = 'valueProperties' | 'referenceProperties';
