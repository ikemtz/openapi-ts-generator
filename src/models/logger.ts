export interface ILogger {
  log: (data: string) => void;
  error: (data: string) => void;
  warn: (data: string) => void;
}

export class MockConsoleLogger implements ILogger {
  // tslint:disable-next-line: no-empty
  public log = (data: string) => {};
  // tslint:disable-next-line: no-empty
  public error = (data: string) => {};
  // tslint:disable-next-line: no-empty
  public warn = (data: string) => {};
}
