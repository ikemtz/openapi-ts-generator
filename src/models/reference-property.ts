export interface IReferenceProperty {
  name: string;
  snakeCaseName: string;
  referenceTypeName: string;
  typeScriptType: string;
  hasValidators: boolean;
  isSameAsParentTypescriptType: boolean;
  hasMultipleValidators: boolean;
  maxItems?: number;
  minItems?: number;
  maxLength?: number;
  minLength?: number;
  maximum?: number;
  minimum?: number;
  isArray: boolean;
  required: boolean;
  isEnum?: boolean;
  initialValue: string;
  initialTestValue: string;
}
