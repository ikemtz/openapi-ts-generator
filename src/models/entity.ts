import { IReferenceProperty } from './reference-property';
import { IValueProperty } from './value-property';
import { IImportType } from './template-data';

export interface IEntity {
  isEnum?: boolean;
  enumValues: (string | { key?: number; name: string; titleName: string })[];
  name: string;
  kebabCasedName: string;
  camelSingularName: string;
  description?: string;
  importTypes: IImportType[];
  valueProperties: IValueProperty[];
  referenceProperties: IReferenceProperty[];
}
