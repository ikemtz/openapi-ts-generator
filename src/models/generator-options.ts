/* eslint-disable @typescript-eslint/no-unused-vars */
import { resolve } from 'path';
import { ILogger } from './logger';
import { IEntity } from './entity';
import { IReferenceProperty } from './reference-property';
import { IValueProperty } from './value-property';

export interface IGeneratorOptions {
  logger?: ILogger;
  outputPath: string;
  openApiJsonUrl?: string;
  openApiJsonFileName?: string;
  genAngularFormGroups?: boolean;
  typeFilterCallBack?: (entity: IEntity, index: number, array: IEntity[]) => boolean;
  valuePropertyTypeFilterCallBack?: (valueProperty: IValueProperty, index: number, array: IValueProperty[]) => boolean;
  referencePropertyTypeFilterCallBack?: (referenceProperty: IReferenceProperty, index: number, array: IReferenceProperty[]) => boolean;
  pathUrlFormattingCallBack?: (url: string) => string;
  templates?: ITemplates | null;
}
export interface ITemplates {
  model?: string;
  testObjectFactory?: string;
  form?: string;
  formGroupFactory?: string;
  modelProperties?: string;
  barrel?: string;
  enum?: string;
  endpoints?: string;
}
export function defaultFilter(
  _value: IEntity | IValueProperty | IReferenceProperty,
  _index: number,
  _array: IEntity[] | IValueProperty[] | IReferenceProperty[],
): boolean {
  return true;
}

export function setGeneratorOptionDefaults(options: IGeneratorOptions): IGeneratorOptions {
  const templateFolder = resolve(`${__dirname}/..`, 'templates');
  options.typeFilterCallBack = options.typeFilterCallBack ?? defaultFilter;
  options.valuePropertyTypeFilterCallBack = options.valuePropertyTypeFilterCallBack ?? defaultFilter;
  options.referencePropertyTypeFilterCallBack = options.referencePropertyTypeFilterCallBack ?? defaultFilter;
  options.templates = {
    ...options.templates,
    model: options.templates?.model ?? `${templateFolder}/model.ts.hbs`,
    form: options.templates?.form ?? `${templateFolder}/form.ts.hbs`,
    formGroupFactory: options.templates?.formGroupFactory ?? `${templateFolder}/form-group-factory.ts.hbs`,
    testObjectFactory: options.templates?.testObjectFactory ?? `${templateFolder}/test-object-factory.ts.hbs`,
    barrel: options.templates?.barrel ?? `${templateFolder}/index.ts.hbs`,
    enum: options.templates?.barrel ?? `${templateFolder}/enum.ts.hbs`,
    modelProperties: options.templates?.barrel ?? `${templateFolder}/model-properties.ts.hbs`,
    endpoints: options.templates?.endpoints ?? `${templateFolder}/endpoints.ts.hbs`,
  };
  return options;
}
