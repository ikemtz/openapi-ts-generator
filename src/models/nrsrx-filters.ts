/* eslint-disable @typescript-eslint/no-unused-vars */
import { IEntity } from './entity';
import { IValueProperty } from './value-property';
const edmRegex = /^Edm[A-z]*Kind$/;

export function nrsrxTypeFilterCallBack(val: IEntity, _i: number, _arr: IEntity[]): boolean {
  return !val.name.endsWith('ODataEnvelope') && !edmRegex.test(val.name);
}
export function nrsrxValuePropertyTypeFilterCallBack(val: IValueProperty, _i: number, _arr: IValueProperty[]): boolean {
  return !val.name.startsWith('created') && !val.name.startsWith('updated');
}
