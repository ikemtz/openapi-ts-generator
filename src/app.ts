import { generateTsModels } from '.';
import { IGeneratorOptions } from './models/generator-options';
import { nrsrxTypeFilterCallBack, nrsrxValuePropertyTypeFilterCallBack } from './models/nrsrx-filters';
import { IEntity } from './models/template-data';

export const unitGenerationOptions = {
  openApiJsonUrl: 'https://d-unit-wal-00-cus-mstrcrp.azurewebsites.net/swagger/v1/swagger.json',
  outputPath: './output_unit/',
  typeFilterCallBack: nrsrxTypeFilterCallBack,
  valuePropertyTypeFilterCallBack: nrsrxValuePropertyTypeFilterCallBack,
};
export const messageGenerationOptions = {
  openApiJsonUrl: 'https://d-msng-wal-01-cus-mstrcrp.azurewebsites.net/swagger/v1/swagger.json',
  outputPath: './output_msng/',
  typeFilterCallBack: (val: IEntity, i: number, arr: IEntity[]) =>
    nrsrxTypeFilterCallBack(val, i, arr) && val.name !== 'GetMessageInfoResponse',
  valuePropertyTypeFilterCallBack: nrsrxValuePropertyTypeFilterCallBack,
};
export const accountGenerationOptions = {
  openApiJsonUrl: 'https://d-acct-wal-01-cus-mstrcrp.azurewebsites.net/swagger/v1/swagger.json',
  outputPath: './output_acct/',
  typeFilterCallBack: nrsrxTypeFilterCallBack,
  valuePropertyTypeFilterCallBack: nrsrxValuePropertyTypeFilterCallBack,
};

export const fileEmployeeGenerationOptions: IGeneratorOptions = {
  openApiJsonFileName: '../src/test-swagger-document.json',
  outputPath: './output_empl/',
  typeFilterCallBack: nrsrxTypeFilterCallBack,
  valuePropertyTypeFilterCallBack: nrsrxValuePropertyTypeFilterCallBack,
};

export function generateUnitFiles(options: IGeneratorOptions = unitGenerationOptions) {
  return generateTsModels(options);
}

export function generateMessageFiles(options: IGeneratorOptions = messageGenerationOptions) {
  return generateTsModels(messageGenerationOptions);
}

export function generateAccountFiles(options: IGeneratorOptions = accountGenerationOptions) {
  return generateTsModels(accountGenerationOptions);
}

export function generateEmployeeFiles(options: IGeneratorOptions = fileEmployeeGenerationOptions) {
  return generateTsModels(fileEmployeeGenerationOptions);
}

generateUnitFiles();
generateMessageFiles();
generateAccountFiles();
generateEmployeeFiles();
