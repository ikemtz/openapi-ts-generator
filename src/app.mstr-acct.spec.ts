import { mkdirSync } from 'fs';
import { generateTsModels, nrsrxTypeFilterCallBack, nrsrxValuePropertyTypeFilterCallBack } from '.';
import { ValidateFiles } from './app.spec';
import { IGeneratorOptions } from './models/generator-options';

const accountGenerationOptionsFactory = (): IGeneratorOptions => ({
  openApiJsonFileName: './open-api-spec-docs/mstr-acct.json',
  outputPath: './jest_output/acct/',
  typeFilterCallBack: nrsrxTypeFilterCallBack,
  valuePropertyTypeFilterCallBack: nrsrxValuePropertyTypeFilterCallBack,
});

describe('Url Based - Full Integration Tests', () => {
  describe('MasterCorp Account Service', () => {
    it('should generate files', async (done) => {
      const options = accountGenerationOptionsFactory();
      try {
        mkdirSync(options.outputPath);
      } catch {
        // ignore
      }
      await generateTsModels(options);
      ValidateFiles(options);
      done();
    });
  });
});
