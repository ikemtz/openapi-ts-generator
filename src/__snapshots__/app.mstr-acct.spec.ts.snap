// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Url Based - Full Integration Tests MasterCorp Account Service should generate files 1`] = `
Array [
  "account-insert-request.form-group-fac.ts",
  "account-insert-request.model.ts",
  "account-insert-request.properties.ts",
  "account-update-request.form-group-fac.ts",
  "account-update-request.model.ts",
  "account-update-request.properties.ts",
  "account.form-group-fac.ts",
  "account.model.ts",
  "account.properties.ts",
  "contact-insert-request.form-group-fac.ts",
  "contact-insert-request.model.ts",
  "contact-insert-request.properties.ts",
  "contact-update-request.form-group-fac.ts",
  "contact-update-request.model.ts",
  "contact-update-request.properties.ts",
  "contact.form-group-fac.ts",
  "contact.model.ts",
  "contact.properties.ts",
  "endpoints.ts",
  "index.ts",
]
`;

exports[`Url Based - Full Integration Tests MasterCorp Account Service should generate files: account.form-group-fac.ts 1`] = `
"/**
 * This file is generated by the openapi-ts-generator
 * #form-group-factory.ts.hbs
 * For issues or feature request, visit the repo: https://github.com/ikemtz/openapi-ts-generator
 * Do not edit.
 */
// eslint-disable-next-line @typescript-eslint/no-unused-vars
import { FormControl, FormArray, FormGroup, Validators } from '@angular/forms'; //NOSONAR

export function AccountFormGroupFac(): FormGroup {
  return new FormGroup({
    id: new FormControl(''),
    refId: new FormControl(''),
    name: new FormControl(''),
    parentAccountId: new FormControl(''),
    isEnabled: new FormControl(''),
    emailInvoice: new FormControl(''),
    tenantId: new FormControl(''),
    parentAccount: new FormControl(''),
    contacts: new FormArray([]),
    inverseParentAccount: new FormArray([]),
  });
};
"
`;

exports[`Url Based - Full Integration Tests MasterCorp Account Service should generate files: account.model.ts 1`] = `
"/**
 * This file is generated by the openapi-ts-generator
 * #model.ts.hbs
 * For issues or feature request, visit the repo: https://github.com/ikemtz/openapi-ts-generator
 * Do not edit.
 */
// eslint-disable-next-line @typescript-eslint/no-empty-interface
import { IContact } from './contact.model';

export interface IAccount {
  id?: string;
  refId?: string;
  name?: string;
  parentAccountId?: string;
  isEnabled?: boolean;
  emailInvoice?: boolean;
  tenantId?: string;
  parentAccount?: IAccount;
  contacts?: IContact[];
  inverseParentAccount?: IAccount[];
}
"
`;

exports[`Url Based - Full Integration Tests MasterCorp Account Service should generate files: account.properties.ts 1`] = `
"/**
 * This file is generated by the openapi-ts-generator
 * #model-properties.ts.hbs
 * For issues or feature request, visit the repo: https://github.com/ikemtz/openapi-ts-generator
 * Do not edit.
 */
/* tslint:disable */
export enum AccountProperties {
  ID = 'id',
  REF_ID = 'refId',
  NAME = 'name',
  PARENT_ACCOUNT_ID = 'parentAccountId',
  IS_ENABLED = 'isEnabled',
  EMAIL_INVOICE = 'emailInvoice',
  TENANT_ID = 'tenantId',
  PARENT_ACCOUNT = 'parentAccount',
  CONTACTS = 'contacts',
  INVERSE_PARENT_ACCOUNT = 'inverseParentAccount',
}
"
`;

exports[`Url Based - Full Integration Tests MasterCorp Account Service should generate files: account-insert-request.form-group-fac.ts 1`] = `
"/**
 * This file is generated by the openapi-ts-generator
 * #form-group-factory.ts.hbs
 * For issues or feature request, visit the repo: https://github.com/ikemtz/openapi-ts-generator
 * Do not edit.
 */
// eslint-disable-next-line @typescript-eslint/no-unused-vars
import { FormControl, FormArray, FormGroup, Validators } from '@angular/forms'; //NOSONAR

export function AccountInsertRequestFormGroupFac(): FormGroup {
  return new FormGroup({
    id: new FormControl(''),
    refId: new FormControl(''),
    name: new FormControl('', Validators.required),
    parentAccountId: new FormControl(''),
    emailInvoice: new FormControl(''),
    contacts: new FormArray([]),
  });
};
"
`;

exports[`Url Based - Full Integration Tests MasterCorp Account Service should generate files: account-insert-request.model.ts 1`] = `
"/**
 * This file is generated by the openapi-ts-generator
 * #model.ts.hbs
 * For issues or feature request, visit the repo: https://github.com/ikemtz/openapi-ts-generator
 * Do not edit.
 */
// eslint-disable-next-line @typescript-eslint/no-empty-interface
import { IContact } from './contact.model';

export interface IAccountInsertRequest {
  id?: string;
  refId?: string;
  name?: string;
  parentAccountId?: string;
  emailInvoice?: boolean;
  contacts?: IContact[];
}
"
`;

exports[`Url Based - Full Integration Tests MasterCorp Account Service should generate files: account-insert-request.properties.ts 1`] = `
"/**
 * This file is generated by the openapi-ts-generator
 * #model-properties.ts.hbs
 * For issues or feature request, visit the repo: https://github.com/ikemtz/openapi-ts-generator
 * Do not edit.
 */
/* tslint:disable */
export enum AccountInsertRequestProperties {
  ID = 'id',
  REF_ID = 'refId',
  NAME = 'name',
  PARENT_ACCOUNT_ID = 'parentAccountId',
  EMAIL_INVOICE = 'emailInvoice',
  CONTACTS = 'contacts',
}
"
`;

exports[`Url Based - Full Integration Tests MasterCorp Account Service should generate files: account-update-request.form-group-fac.ts 1`] = `
"/**
 * This file is generated by the openapi-ts-generator
 * #form-group-factory.ts.hbs
 * For issues or feature request, visit the repo: https://github.com/ikemtz/openapi-ts-generator
 * Do not edit.
 */
// eslint-disable-next-line @typescript-eslint/no-unused-vars
import { FormControl, FormArray, FormGroup, Validators } from '@angular/forms'; //NOSONAR

export function AccountUpdateRequestFormGroupFac(): FormGroup {
  return new FormGroup({
    id: new FormControl(''),
    refId: new FormControl('', Validators.required),
    name: new FormControl('', Validators.required),
    parentAccountId: new FormControl(''),
    isEnabled: new FormControl(''),
    emailInvoice: new FormControl(''),
    contacts: new FormArray([]),
  });
};
"
`;

exports[`Url Based - Full Integration Tests MasterCorp Account Service should generate files: account-update-request.model.ts 1`] = `
"/**
 * This file is generated by the openapi-ts-generator
 * #model.ts.hbs
 * For issues or feature request, visit the repo: https://github.com/ikemtz/openapi-ts-generator
 * Do not edit.
 */
// eslint-disable-next-line @typescript-eslint/no-empty-interface
import { IContact } from './contact.model';

export interface IAccountUpdateRequest {
  id?: string;
  refId?: string;
  name?: string;
  parentAccountId?: string;
  isEnabled?: boolean;
  emailInvoice?: boolean;
  contacts?: IContact[];
}
"
`;

exports[`Url Based - Full Integration Tests MasterCorp Account Service should generate files: account-update-request.properties.ts 1`] = `
"/**
 * This file is generated by the openapi-ts-generator
 * #model-properties.ts.hbs
 * For issues or feature request, visit the repo: https://github.com/ikemtz/openapi-ts-generator
 * Do not edit.
 */
/* tslint:disable */
export enum AccountUpdateRequestProperties {
  ID = 'id',
  REF_ID = 'refId',
  NAME = 'name',
  PARENT_ACCOUNT_ID = 'parentAccountId',
  IS_ENABLED = 'isEnabled',
  EMAIL_INVOICE = 'emailInvoice',
  CONTACTS = 'contacts',
}
"
`;

exports[`Url Based - Full Integration Tests MasterCorp Account Service should generate files: contact.form-group-fac.ts 1`] = `
"/**
 * This file is generated by the openapi-ts-generator
 * #form-group-factory.ts.hbs
 * For issues or feature request, visit the repo: https://github.com/ikemtz/openapi-ts-generator
 * Do not edit.
 */
// eslint-disable-next-line @typescript-eslint/no-unused-vars
import { FormControl, FormArray, FormGroup, Validators } from '@angular/forms'; //NOSONAR

export function ContactFormGroupFac(): FormGroup {
  return new FormGroup({
    id: new FormControl(''),
    isEnabled: new FormControl(''),
    contactTypeId: new FormControl('', Validators.required),
    accountId: new FormControl('', Validators.required),
    firstName: new FormControl('', Validators.required),
    lastName: new FormControl('', Validators.required),
    middleName: new FormControl(''),
    addressLine1: new FormControl('', Validators.required),
    addressLine2: new FormControl(''),
    city: new FormControl('', Validators.required),
    state: new FormControl('', Validators.required),
    zip: new FormControl('', Validators.required),
    emails: new FormControl(''),
    phone1: new FormControl(''),
    phone2: new FormControl(''),
    account: new FormControl(''),
  });
};
"
`;

exports[`Url Based - Full Integration Tests MasterCorp Account Service should generate files: contact.model.ts 1`] = `
"/**
 * This file is generated by the openapi-ts-generator
 * #model.ts.hbs
 * For issues or feature request, visit the repo: https://github.com/ikemtz/openapi-ts-generator
 * Do not edit.
 */
// eslint-disable-next-line @typescript-eslint/no-empty-interface
import { IAccount } from './account.model';

export interface IContact {
  id?: string;
  isEnabled?: boolean;
  contactTypeId?: string;
  accountId?: string;
  firstName?: string;
  lastName?: string;
  middleName?: string;
  addressLine1?: string;
  addressLine2?: string;
  city?: string;
  state?: string;
  zip?: string;
  emails?: string;
  phone1?: string;
  phone2?: string;
  account?: IAccount;
}
"
`;

exports[`Url Based - Full Integration Tests MasterCorp Account Service should generate files: contact.properties.ts 1`] = `
"/**
 * This file is generated by the openapi-ts-generator
 * #model-properties.ts.hbs
 * For issues or feature request, visit the repo: https://github.com/ikemtz/openapi-ts-generator
 * Do not edit.
 */
/* tslint:disable */
export enum ContactProperties {
  ID = 'id',
  IS_ENABLED = 'isEnabled',
  CONTACT_TYPE_ID = 'contactTypeId',
  ACCOUNT_ID = 'accountId',
  FIRST_NAME = 'firstName',
  LAST_NAME = 'lastName',
  MIDDLE_NAME = 'middleName',
  ADDRESS_LINE_1 = 'addressLine1',
  ADDRESS_LINE_2 = 'addressLine2',
  CITY = 'city',
  STATE = 'state',
  ZIP = 'zip',
  EMAILS = 'emails',
  PHONE_1 = 'phone1',
  PHONE_2 = 'phone2',
  ACCOUNT = 'account',
}
"
`;

exports[`Url Based - Full Integration Tests MasterCorp Account Service should generate files: contact-insert-request.form-group-fac.ts 1`] = `
"/**
 * This file is generated by the openapi-ts-generator
 * #form-group-factory.ts.hbs
 * For issues or feature request, visit the repo: https://github.com/ikemtz/openapi-ts-generator
 * Do not edit.
 */
// eslint-disable-next-line @typescript-eslint/no-unused-vars
import { FormControl, FormArray, FormGroup, Validators } from '@angular/forms'; //NOSONAR

export function ContactInsertRequestFormGroupFac(): FormGroup {
  return new FormGroup({
    id: new FormControl(''),
    contactTypeId: new FormControl('', Validators.required),
    accountId: new FormControl('', Validators.required),
    firstName: new FormControl('', Validators.required),
    lastName: new FormControl('', Validators.required),
    middleName: new FormControl(''),
    addressLine1: new FormControl('', Validators.required),
    addressLine2: new FormControl(''),
    city: new FormControl('', Validators.required),
    state: new FormControl('', Validators.required),
    zip: new FormControl('', Validators.required),
    emails: new FormControl(''),
    phone1: new FormControl(''),
    phone2: new FormControl(''),
  });
};
"
`;

exports[`Url Based - Full Integration Tests MasterCorp Account Service should generate files: contact-insert-request.model.ts 1`] = `
"/**
 * This file is generated by the openapi-ts-generator
 * #model.ts.hbs
 * For issues or feature request, visit the repo: https://github.com/ikemtz/openapi-ts-generator
 * Do not edit.
 */
// eslint-disable-next-line @typescript-eslint/no-empty-interface
export interface IContactInsertRequest {
  id?: string;
  contactTypeId?: string;
  accountId?: string;
  firstName?: string;
  lastName?: string;
  middleName?: string;
  addressLine1?: string;
  addressLine2?: string;
  city?: string;
  state?: string;
  zip?: string;
  emails?: string;
  phone1?: string;
  phone2?: string;
}
"
`;

exports[`Url Based - Full Integration Tests MasterCorp Account Service should generate files: contact-insert-request.properties.ts 1`] = `
"/**
 * This file is generated by the openapi-ts-generator
 * #model-properties.ts.hbs
 * For issues or feature request, visit the repo: https://github.com/ikemtz/openapi-ts-generator
 * Do not edit.
 */
/* tslint:disable */
export enum ContactInsertRequestProperties {
  ID = 'id',
  CONTACT_TYPE_ID = 'contactTypeId',
  ACCOUNT_ID = 'accountId',
  FIRST_NAME = 'firstName',
  LAST_NAME = 'lastName',
  MIDDLE_NAME = 'middleName',
  ADDRESS_LINE_1 = 'addressLine1',
  ADDRESS_LINE_2 = 'addressLine2',
  CITY = 'city',
  STATE = 'state',
  ZIP = 'zip',
  EMAILS = 'emails',
  PHONE_1 = 'phone1',
  PHONE_2 = 'phone2',
}
"
`;

exports[`Url Based - Full Integration Tests MasterCorp Account Service should generate files: contact-update-request.form-group-fac.ts 1`] = `
"/**
 * This file is generated by the openapi-ts-generator
 * #form-group-factory.ts.hbs
 * For issues or feature request, visit the repo: https://github.com/ikemtz/openapi-ts-generator
 * Do not edit.
 */
// eslint-disable-next-line @typescript-eslint/no-unused-vars
import { FormControl, FormArray, FormGroup, Validators } from '@angular/forms'; //NOSONAR

export function ContactUpdateRequestFormGroupFac(): FormGroup {
  return new FormGroup({
    id: new FormControl(''),
    contactTypeId: new FormControl('', Validators.required),
    accountId: new FormControl('', Validators.required),
    firstName: new FormControl('', Validators.required),
    lastName: new FormControl('', Validators.required),
    middleName: new FormControl(''),
    addressLine1: new FormControl('', Validators.required),
    addressLine2: new FormControl(''),
    city: new FormControl('', Validators.required),
    state: new FormControl('', Validators.required),
    zip: new FormControl('', Validators.required),
    emails: new FormControl(''),
    phone1: new FormControl(''),
    phone2: new FormControl(''),
  });
};
"
`;

exports[`Url Based - Full Integration Tests MasterCorp Account Service should generate files: contact-update-request.model.ts 1`] = `
"/**
 * This file is generated by the openapi-ts-generator
 * #model.ts.hbs
 * For issues or feature request, visit the repo: https://github.com/ikemtz/openapi-ts-generator
 * Do not edit.
 */
// eslint-disable-next-line @typescript-eslint/no-empty-interface
export interface IContactUpdateRequest {
  id?: string;
  contactTypeId?: string;
  accountId?: string;
  firstName?: string;
  lastName?: string;
  middleName?: string;
  addressLine1?: string;
  addressLine2?: string;
  city?: string;
  state?: string;
  zip?: string;
  emails?: string;
  phone1?: string;
  phone2?: string;
}
"
`;

exports[`Url Based - Full Integration Tests MasterCorp Account Service should generate files: contact-update-request.properties.ts 1`] = `
"/**
 * This file is generated by the openapi-ts-generator
 * #model-properties.ts.hbs
 * For issues or feature request, visit the repo: https://github.com/ikemtz/openapi-ts-generator
 * Do not edit.
 */
/* tslint:disable */
export enum ContactUpdateRequestProperties {
  ID = 'id',
  CONTACT_TYPE_ID = 'contactTypeId',
  ACCOUNT_ID = 'accountId',
  FIRST_NAME = 'firstName',
  LAST_NAME = 'lastName',
  MIDDLE_NAME = 'middleName',
  ADDRESS_LINE_1 = 'addressLine1',
  ADDRESS_LINE_2 = 'addressLine2',
  CITY = 'city',
  STATE = 'state',
  ZIP = 'zip',
  EMAILS = 'emails',
  PHONE_1 = 'phone1',
  PHONE_2 = 'phone2',
}
"
`;

exports[`Url Based - Full Integration Tests MasterCorp Account Service should generate files: endpoints.ts 1`] = `
"/**
 * This file is generated by the openapi-ts-generator
 * #endpoints-ts.hbs
 * For issues or feature request, visit the repo: https://github.com/ikemtz/openapi-ts-generator
 * Do not edit.
 */
/* tslint:disable */
export enum Endpoints {
  accounts = '/api/v1/Accounts.{format}',
  contacts = '/api/v1/Contacts.{format}',
}
export interface IEndpoints {
  accounts: string;
  contacts: string;
}
"
`;

exports[`Url Based - Full Integration Tests MasterCorp Account Service should generate files: index.ts 1`] = `
"/**
 * This file is generated by the openapi-ts-generator
 * #generate-barrel-ts.hbs
 * For issues or feature request, visit the repo: https://github.com/ikemtz/openapi-ts-generator
 * Do not edit.
 */
/* tslint:disable */
export * from './account-insert-request.form-group-fac';
export * from './account-insert-request.model';
export * from './account-insert-request.properties';
export * from './account-update-request.form-group-fac';
export * from './account-update-request.model';
export * from './account-update-request.properties';
export * from './account.form-group-fac';
export * from './account.model';
export * from './account.properties';
export * from './contact-insert-request.form-group-fac';
export * from './contact-insert-request.model';
export * from './contact-insert-request.properties';
export * from './contact-update-request.form-group-fac';
export * from './contact-update-request.model';
export * from './contact-update-request.properties';
export * from './contact.form-group-fac';
export * from './contact.model';
export * from './contact.properties';
export * from './endpoints';
"
`;
